java/vector/src/main/java/org/apache/arrow/vector/stream/MessageSerializer.java

```diff
@@ -120,6 +120,9 @@ public static ArrowBlock serialize(WriteChannel out, ArrowRecordBatch batch)
 
     long start = out.getCurrentPosition();
     int bodyLength = batch.computeBodyLength();
+    if (bodyLength % 8 != 0) {
+      bodyLength += 8 - (bodyLength % 8);
+    }
```

Comment:
this code here is what makes me think buffers aren't always multiples of 8:
https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/schema/ArrowRecordBatch.java#L78
The problem is that that code aligns the *next* buffer to write, but if it's the last buffer it doesn't get padded at the end, which is what this code block is attempting to do